internal/controller/children.go

```diff
@@ -155,7 +155,8 @@ func NewChildResources(cr *amaltheadevv1alpha1.AmaltheaSession) ChildResources {
 
 	desiredDataSourcesPVCs := []ChildResource[v1.PersistentVolumeClaim]{}
 	specPVCs, _, _ := cr.DataSources()
-	for _, desiredPVC := range specPVCs {
+	for i := range specPVCs {
+		desiredPVC := specPVCs[i]
```

Comment:
Note: This makes a copy (which does not seem needed), but is valid code. So no need to change it to use pointers.

Reply:
Yeah that makes sense.